extensions/wikia/PageHeader/Languages.class.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Wikia\PageHeader;
+
+use \RequestContext;
+use \WikiaApp;
+
+class Languages {
+
+	public $currentLangName;
+	public $languageList;
+	private $title;
+
+	/**
+	 * Language constructor.
+	 *
+	 * @param WikiaApp $app
+	 */
+	public function __construct( WikiaApp $app ) {
+		$this->currentLangName = \Language::getLanguageName( RequestContext::getMain()->getLanguage()->getCode() );
+		$this->languageList = $this->handleLanguages( $app );
+		$this->title = $app->wg->title;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)